pallets/manta-sbt/src/lib.rs

```diff
@@ -0,0 +1,1029 @@
+    transactional, PalletId,
+};
+use frame_system::pallet_prelude::*;
+use manta_support::manta_pay::{
+    asset_value_encode, fp_decode, fp_encode, id_from_field, AssetValue, Checkpoint,
+    FullIncomingNote, MTParametersError, PullResponse, ReceiverChunk, SenderChunk, StandardAssetId,
+    TransferPost, Utxo, UtxoAccumulatorOutput, UtxoItemHashError, UtxoMerkleTreePath,
+    VerifyingContextError, Wrap, WrapPair,
+};
+use sp_runtime::{
+    traits::{AccountIdConversion, One},
+    ArithmeticError,
+};
+
+use manta_pay::{
+    config::{self, utxo::MerkleTreeConfiguration},
+    manta_accounting::transfer::{
+        self,
+        canonical::TransferShape,
+        receiver::{ReceiverLedger, ReceiverPostError},
+        sender::{SenderLedger, SenderPostError},
+        InvalidAuthorizationSignature, InvalidSinkAccount, InvalidSourceAccount, ProofSystemError,
+        SinkPostingKey, SourcePostingKey, TransferLedger, TransferLedgerSuperPostingKey,
+        TransferPostingKeyRef,
+    },
+    manta_crypto::merkle_tree::{self, forest::Configuration as _},
+    manta_parameters::{self, Get as _},
+    manta_util::codec::Decode as _,
+    parameters::load_transfer_parameters,
+};
+use manta_util::codec::Encode;
+
+pub use pallet::*;
+pub use weights::WeightInfo;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+pub mod weights;
+
+#[cfg(feature = "runtime-benchmarks")]
+pub mod benchmark;
+
+#[cfg(feature = "rpc")]
+pub mod rpc;
+
+#[cfg(feature = "runtime")]
+pub mod runtime;
+
+// One in encoded form, used to check that value input in `ToPrivate` post is one
+const ENCODED_ONE: [u8; 16] = 1u128.to_le_bytes();
+
+/// Type alias for currency balance.
+pub type BalanceOf<T> =
+    <<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+
+/// MantaSBT Pallet
+#[frame_support::pallet]
+pub mod pallet {
+    use super::*;
+
+    const STORAGE_VERSION: StorageVersion = StorageVersion::new(1);
+
+    /// Pallet
+    #[pallet::pallet]
+    #[pallet::generate_store(pub(super) trait Store)]
+    #[pallet::storage_version(STORAGE_VERSION)]
+    pub struct Pallet<T>(_);
+
+    /// The module configuration trait.
+    #[pallet::config]
+    pub trait Config: frame_system::Config {
+        /// The overarching event type.
+        type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+        /// Weight information for extrinsics in this pallet.
+        type WeightInfo: WeightInfo;
+
+        /// The currency mechanism.
+        type Currency: ReservableCurrency<Self::AccountId>;
+
+        /// Pallet ID
+        type PalletId: Get<PalletId>;
+
+        /// Number of unique Asset Ids reserved per `reserve_sbt` call, is the amount of SBTs allowed to be minted
+        #[pallet::constant]
+        type MintsPerReserve: Get<u16>;
+
+        /// Price to reserve Asset Ids
+        type ReservePrice: Get<BalanceOf<Self>>;
```

Comment:
Can't the palletid and reserveprice be #[pallet::constant] ?

Reply:
yes they should be